src/liballoc/sync.rs

```diff
@@ -751,6 +751,46 @@ impl<T: ?Sized> Arc<T> {
         this.inner().strong.load(SeqCst)
     }
 
+    /// Increment the strong count by 1.
+    #[inline]
+    #[unstable(feature = "arc_mutate_count", issue = "0")]
+    pub unsafe fn increment_strong_count(this: &Self) {
```

Comment:
We could make this function safe, I think - seems like not a bad idea, as it's a somewhat common use case.

Reply:
You're right that that *should* be safe given it's equivalent to `mem::forget(Arc::clone(&value))`. I was going off https://github.com/rust-lang/rust/pull/68700#discussion_r396169064 which mentions:

> "TBH just having unsafe methods to fiddle with Arc's ref count would be nicer than any of this"

This method should probably be used cautiously, and we should provide clear guidance. But I guess from a _safety_ perspective there's not a clear reason to make this unsafe.

Reply:
Alright, gone and turned this into a safe method since at worst this would leak, which is not `unsafe`.